ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -8295,6 +8295,45 @@ template <> void init_pushconst_tensor_offsets(ggml_backend_vk_context * ctx, vk
     GGML_UNUSED(src2);
 }
 
+static vk_subbuffer ggml_vk_tensor_subbuffer(
+    const ggml_backend_vk_context * ctx, const ggml_tensor * tensor, bool support_incontiguous,
+    vk_buffer buffer = nullptr, size_t offset = 0) {
+
+    if (!buffer) {
+        auto buf_ctx = (ggml_backend_vk_buffer_context *)tensor->buffer->context;
+        buffer = buf_ctx->dev_buffer;
+        offset = vk_tensor_offset(tensor) + tensor->view_offs;
+    }
+    GGML_ASSERT(buffer != nullptr);
+
+    size_t misalign_bytes = offset & (ctx->device->properties.limits.minStorageBufferOffsetAlignment - 1);
+    offset &= ~misalign_bytes;
+
+    size_t size;
+    if (support_incontiguous) {
+        size = ggml_nbytes(tensor) + misalign_bytes;
+        if (offset + size >= buffer->size) {
+            size = ggml_vk_get_max_buffer_range(ctx, buffer, offset);
+        }
+    } else {
+        size_t elem_size = ggml_type_size(tensor->type) / ggml_blck_size(tensor->type);
```

Comment:
This replicates the branch here: https://github.com/ggml-org/llama.cpp/blob/master/ggml/src/ggml-vulkan/ggml-vulkan.cpp#L8632-L8636

From what I understand it assumes that the OP doesn't handle non-contiguous, so all tensors are expected to be contiguous. But in that case this just ends up being equivalent to `ggml_nbytes(tensor)`, which is the same as without this branch?

NB: the original code ignores block size for sources other than src0, but that just seems dangorous at best